sklearn/tree/_tree.pyx

```diff
@@ -224,6 +226,8 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
         cdef stack[StackRecord] builder_stack
         cdef StackRecord stack_record
 
+        cdef ParentInfo parent_record
```

Comment:
`_init_parent_record` need to be called here to have the same behavior as `main`?:

```suggestion
        cdef ParentInfo parent_record
        _init_parent_record(&parent_record)
```

Reply:
Done. 
d0a8d2f